### ISSUE TITLE
OutOfBoundsError when drawing shapes with embedded_graphics

### ISSUE THREAD

**Issue by TravisThomp** [IniterWorker/gc9a01#11]
I'm using a raspberry pi zero-2w and the rppal bindings.

My setup is 
```rust
let interface = SPIDisplayInterface::new(spi_device, dc);
let display = Gc9a01::new(interface, DisplayResolution240x240, self.rotation).into_buffered_graphics();
```

When I draw with `display.set_pixel(...)` it works perfectly,
but when I attempt to use embedded_graphics to draw a simple rectangle(or any shape)
```rust
let thin_stroke = PrimitiveStyle::with_fill(Rgb565::GREEN);
match Rectangle::new(Point::new(16, 16), Size::new(20, 20)).into_styled(thin_stroke).draw(&mut self.display)
{
    Ok(_) => println!("Drew it!"),
    Err(e) => println!("Failed to draw rectangle: {:?}", e),
}
self.display.flush()?;
```

I get this error `Failed to draw rectangle: OutOfBoundsError`

I've seen only code that works based off `lvgl` but I'm not in need of UI. Is there something I'm missing to get this to work with just embedded_graphics?

**Comment by IniterWorker** [comment]
Thanks for reporting the regression. I believe that the changes in [commit 37445e33](https://github.com/IniterWorker/gc9a01/commit/37445e33a9a01065627f0ab9cb2281c8b3e31e4c#diff-fd87de66c2a6872baad7f231a2a8c7dc0ac58837f4ce3fe058cd5de4539f35daR202) should resolve the issue.

Please test it and let me know if this solves the issue, or if there are any further problems.

**Comment by IniterWorker** [comment]
Fixed on `0.4.2`. the version `0.4.1` is yanked.
Feel free to reopen if the problem still exists.

**Comment by TravisThomp** [comment]
Works like a charm, thank you! :)
